File: demo/src/main/java/com/backend/demo/entity/Item.java
package com.backend.demo.entity;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import com.fasterxml.jackson.annotation.JsonBackReference;

@Entity
@Table(name="item")
public class Item {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id")
	private int id;
	
	@Column(name="name")
	private String name;
	
	@Column(name="quantity")
	private  int quantity;
	
	@Column(name="price")
	private double price;
	
//	@ManyToOne
//	@JoinColumn(name="shipment_id",nullable = true)
//	@JsonBackReference
//	@OnDelete(action = OnDeleteAction.CASCADE)
	@OneToOne(mappedBy = "item",cascade = CascadeType.ALL)
	@JsonBackReference
	private Shipment shipment;
	
	@Column(name = "weight")
	private String weight;
	
	@Column(name = "description")
	private String description;
	
	@Column(name = "refrence_number")
	private String refrenceNumber;
	
	public Item() {}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public int getQuantity() {
		return quantity;
	}

	public void setQuantity(int quantity) {
		this.quantity = quantity;
	}

	public double getPrice() {
		return price;
	}

	public void setPrice(double price) {
		this.price = price;
	}

	public Shipment getShipment() {
		return shipment;
	}

	public void setShipment(Shipment shipment) {
		this.shipment = shipment;
	}

	public String getWeight() {
		return weight;
	}

	public void setWeight(String weight) {
		this.weight = weight;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getRefrenceNumber() {
		return refrenceNumber;
	}

	public void setRefrenceNumber(String refrenceNumber) {
		this.refrenceNumber = refrenceNumber;
	}
	
	
	
	

}

File: delivery_agent_app/lib/models/Item.dart
class Item {
  final int? id;
  final String? name;
  final int? quantity;
  final double? price;
  final String? weight;
  final String? description;
  final String? referenceNumber;

  Item({
    required this.id,
    required this.name,
    required this.price,
    required this.quantity,
    required this.weight,
    required this.description,
    required this.referenceNumber,
  });

  factory Item.fromJson(json) {
    return Item(
      id: json['id'],
      name: json['name'] ?? '',
      price: json['price'] ?? 0,
      quantity: json['quantity'] ?? 1,
      weight: json['weight'] ?? 1,
      description: json['description'] ?? '',
      referenceNumber: json['referenceNumber'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name ?? '',
      'price': price ?? 0,
      'quantity': quantity ?? 1,
      'weight': weight ?? 1,
      'description': description ?? '',
      'referenceNumber': referenceNumber ?? '',
    };
  }
}
